includes/class-wc-subscription.php

```diff
@@ -1913,12 +1913,18 @@ public function payment_failed( $new_status = 'on-hold' ) {
 		$this->add_order_note( __( 'Payment failed.', 'woocommerce-subscriptions' ) );
 
 		// Allow a short circuit for plugins & payment gateways to force max failed payments exceeded
-		if ( 'cancelled' == $new_status || apply_filters( 'woocommerce_subscription_max_failed_payments_exceeded', false, $this ) ) {
-			if ( $this->can_be_updated_to( 'cancelled' ) ) {
-				$this->update_status( 'cancelled', __( 'Subscription Cancelled: maximum number of failed payments reached.', 'woocommerce-subscriptions' ) );
+		// This also forces the new status to be 'cancelled' if the filter is applied or the subscription is pending-cancel
+		$status_note                                 = '';
+		$max_failed_payments_exceeded_filter_applied = apply_filters( 'woocommerce_subscription_max_failed_payments_exceeded', false, $this );
```

Comment:
Yeah, I agree. Usually, it is best practice to name variables according to exactly what they are in PHP (citing PHP here because in Go, for example, it would be OK to name this `mfp`), even if it gets long. So, personally, my issue is more with the attribution alignment rule than with the variable name itself 😅 , but I understand this is a global rule. Change applied ✅ 

I loved your suggestion! I think it is simpler to understand 👍 . Also applied